cli/wallet/wallet.go

```diff
@@ -547,22 +547,49 @@ func importMultisig(ctx *cli.Context) error {
 		l := ctx.String("name")
 		label = &l
 	}
-	acc, err := newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
-	if err != nil {
-		return cli.NewExitError(err, 1)
-	}
 
-	if err := acc.ConvertMultisig(m, pubs); err != nil {
-		return cli.NewExitError(err, 1)
-	}
+	var acc *wallet.Account
 
+	// Check if a WIF is provided, and if so, create an account from it
+	if ctx.IsSet("wif") {
+		acc, err = newAccountFromWIF(ctx.App.Writer, ctx.String("wif"), wall.Scrypt, label, pass)
+		if err != nil {
+			return cli.NewExitError(err, 1)
+		}
+	} else {
+		// Otherwise, check if any of the provided public keys correspond to an existing key in the wallet
+		for _, pub := range pubs {
+			existAccount, err := findAccountByPublicKey(wall, pub)
+			if err == nil {
+				acc = existAccount
+				break
+			}
+		}
+	}
+	if acc != nil {
+		// Convert the account to multisig
+		if err := acc.ConvertMultisig(m, pubs); err != nil {
+			return cli.NewExitError(err, 1)
+		}
```

Comment:
Have you tested it? I'd expect it to fail because account is not decrypted and `ConvertMultisig` needs decrypted private key. You need to write your own code for converting encrypted account to multisig.